src/adapters/ip_camera/upnp_listener.rs

```diff
@@ -54,13 +54,12 @@ impl UpnpListener for IpCameraUpnpListener {
 
         let url = try_get!(service.description, "/root/device/presentationURL");
 
-        let mut udn = try_get!(service.description, "/root/device/UDN").clone();
+        let udn = try_get!(service.description, "/root/device/UDN")
         // The UDN is typically of the for uuid:SOME-UID-HERE, but some devices
         // response with just a UUID. We strip off the uuid: prefix, if it exists
         // and use the resulting UUID as the service id.
-        if udn.starts_with("uuid:") {
-            udn = String::from(&udn[5..]);
-        }
+            .trim_left_matches("uuid:")
+            .to_owned();
```

Comment:
I filed a bug with clippy, and it's been fixed in the latest version (not sure when it will be released).
See: https://github.com/Manishearth/rust-clippy/issues/975

Looks like it should be fixed in release 0.0.72